apps/dotcms-ui/src/app/portlets/shared/dot-content-types-listing/components/dot-content-type-clone-dialog/dot-content-type-clone-dialog.component.spec.ts

```diff
@@ -0,0 +1,133 @@
+                declarations: [DotContentTypeCloneDialogComponent, TestHostComponent],
+                imports: [
+                    DotFormSelectorModule,
+                    BrowserAnimationsModule,
+                    DotFieldValidationMessageModule,
+                    DotMdIconSelectorModule,
+                    SiteSelectorFieldModule,
+                    DotDialogModule
+                ],
+                providers: [{ provide: SiteService, useValue: siteServiceMock }]
+                //schemas: [CUSTOM_ELEMENTS_SCHEMA]
+            });
+            fixture = DOTTestBed.createComponent(TestHostComponent);
+            de = fixture.debugElement.query(By.css('dot-content-type-clone-dialog'));
+            component = de.componentInstance;
+            fixture.detectChanges();
+            dotdialog = de.query(By.css('dot-dialog'));
+
+            component.isVisibleDialog = true;
+            fixture.detectChanges();
+        })
+    );
+
+    it('should create', () => {
+        expect(component).toBeTruthy();
+    });
+    it('should have a form', () => {
+        fixture.detectChanges();
+        const form: DebugElement = de.query(By.css('form'));
+        expect(form).not.toBeNull();
+        expect(component.form).toEqual(form.componentInstance.form);
+    });
+    it('should be invalid if no name was added', () => {
+        fixture.detectChanges();
+        expect(component.form.valid).toEqual(false);
+    });
+    it('should be valid if bundle field is added', () => {
+        fixture.detectChanges();
+
+        component.form.setValue(formValues);
+        expect(component.form.valid).toEqual(true);
+    });
+    it('should call cancelBtn() on cancel button click', async () => {
+        const cancelButton: DebugElement = dotdialog.query(By.css('.dialog__button-cancel'));
+        expect(cancelButton).toBeDefined();
+
+        spyOn(component, 'closeDialog');
+
+        cancelButton.nativeElement.click();
+
+        expect(component.closeDialog).toHaveBeenCalledTimes(1);
+
+        component.cancelBtn.subscribe((res) => {
+            expect(res).toEqual(true);
+        });
+    });
+
+    it('should call submitForm() on Copy button click and form valid', async () => {
+        const copyButton: DebugElement = dotdialog.query(By.css('.dialog__button-accept'));
+        expect(copyButton).toBeDefined();
+
+        component.form.setValue(formValues);
+        expect(component.form.valid).toEqual(true);
+
+        spyOn(component, 'submitForm');
+
+        fixture.detectChanges();
+
+        copyButton.nativeElement.click();
+
+        expect(component.submitForm).toHaveBeenCalledTimes(1);
+    });
+
+    it("shouldn't call submitForm() on Copy button click and form invalid", async () => {
+        const copyButton: DebugElement = dotdialog.query(By.css('.dialog__button-accept'));
+        expect(copyButton).toBeDefined();
+
+        expect(component.form.valid).toEqual(false);
+        spyOn(component, 'submitForm');
```

Comment:
spy instead the output of the `dot-test-host-component` not the own component.

Reply:
done!